File: BinarySearch/BinarySearch.java
class Solution {
    public int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;
        while (left <= right) {
            int m = left + (right - left) / 2;
            if (nums[m] == target) {
                return m;
            } else if (target < nums[m]) {
                right = m - 1;
            } else {
                left = m + 1;
            }
        }
        return -1;
    }
}

File: BinarySearch/BinarySearch.py
class Solution:
    def search(self, nums, target):
        left = 0
        right = len(nums) - 1
        
        while left <= right:
            m = left + (right - left) // 2
            if nums[m] == target:
                return m
            elif target < nums[m]:
                right = m - 1
            else:
                left = m + 1
        
        return -1
